### PR TITLE
Use mlir transform dialect Python bindings.

### PR DESCRIPTION
**Pull request by matthias-springer**
The IREE transform dialect Python bindings are removed.

### CODE REVIEW

File: llvm-external-projects/iree-dialects/python/iree/compiler/dialects/_iree_linalg_transform_ops_ext.py [hunk 1]
```diff
@@ -1,111 +0,0 @@
...
-
-def _count_expected_loops(tile_sizes: ir.ArrayAttr) -> int:
-  # Number of loops = number of tile sizes != 0
-  zero = _ensure_int_attr(0)
-  return len(list(tile_sizes)) - list(tile_sizes).count(zero)
-
-
-##===----------------------------------------------------------------------===##
-## LinalgExt specific transforms
-##===----------------------------------------------------------------------===##
```
Comment (ftynse) [inline llvm-external-projects/iree-dialects/python/iree/compiler/dialects/_iree_linalg_transform_ops_ext.py:85]: This looks IREE-specific and does not exist upstream.

File: llvm-external-projects/iree-dialects/python/iree/compiler/dialects/iree_linalg_transform.py [hunk 2]
```diff
@@ -1,8 +0,0 @@
-# Copyright 2021 The IREE Authors
-#
-# Licensed under the Apache License v2.0 with LLVM Exceptions.
-# See https://llvm.org/LICENSE.txt for license information.
-# SPDX-License-Identifier: Apache-2.0 WITH LLVM-exception
-
-from ._iree_linalg_transform_ops_gen import *
-from .._mlir_libs._ireeDialects.iree_linalg_transform import *
```
Comment (ftynse) [inline llvm-external-projects/iree-dialects/python/iree/compiler/dialects/iree_linalg_transform.py:8]: This does not exist upstream.

File: llvm-external-projects/iree-dialects/python/iree/compiler/dialects/_transform_ops_ext.py [hunk 3]
```diff
@@ -1,465 +0,0 @@
...
-    super().__init__(_ensure_bool_attr(reassociate_fp_reductions, False),
-                     _ensure_bool_attr(enable_index_optimizations, False),
-                     _ensure_bool_attr(enable_arm_neon, False),
-                     _ensure_bool_attr(enable_arm_sve, False),
-                     _ensure_bool_attr(enable_amx, False),
-                     _ensure_bool_attr(enable_x86vector, False),
-                     _ensure_bool_attr(enable_async, False),
-                     loc=loc,
-                     ip=ip)
-
```
Comment (ftynse) [inline llvm-external-projects/iree-dialects/python/iree/compiler/dialects/_transform_ops_ext.py:217]: These two aren't upstream

**Review by ftynse — COMMENTED** [review]
Some of the things that are being removed are not available upstream. If this doesn't break any tests, we may be missing coverage.

File: llvm-external-projects/iree-dialects/python/iree/compiler/dialects/_iree_linalg_transform_ops_ext.py (hunk 1 above)
Comment (matthias-springer) [reply llvm-external-projects/iree-dialects/python/iree/compiler/dialects/_iree_linalg_transform_ops_ext.py]: Leaving this as is for now.

**Review by ftynse — APPROVED** [review]

**Pull request merged**